kubectl-metalk8s/pkg/cmd/listNodes.go

```diff
@@ -0,0 +1,106 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	"k8s.io/klog"
+)
+
+type ListNodesOptions struct {
+	config *rest.Config
+
+	genericclioptions.IOStreams
+}
+
+// NewListNodesOptions provides an instance of ListNodesOptions with default values
+func NewListNodesOptions(streams genericclioptions.IOStreams) *ListNodesOptions {
+	return &ListNodesOptions{
+		IOStreams: streams,
+	}
+}
+
+// NewCmdListNodes provide a cobra command wrapping ListNodesOptions
+func NewCmdListNodes(streams genericclioptions.IOStreams) *cobra.Command {
+	opt := NewListNodesOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:          "list-nodes",
+		Short:        "Test list all Kubernetes nodes",
+		SilenceUsage: true,
```

Comment:
Why?

Reply:
Because if you do not silence when `RunE` return an error it will show usage even if the error is not related to some arguments.

So either we silence usage or we handle error output ourselves and exit ourselves when running the command. (So basically mean here we always `return nil` in Run function)

Reply:
Note that my approach is the same as sample-cli-plugin one (kubectl has a special function to check for err and log it + Exit)

So both doable

Personally I don't really like the "usage" output when you give invalid keyword, argument or whatever but it's only my opinion :smiley:

Reply:
The problem is indeed when `RunE` returns a non-nil `error`, Cobra will show usage.

However, there's a way around to *both* show 'usage' when passing invalid arguments and such, as well as not show 'usage' when input validation has been done: https://github.com/scality/metalk8s/compare/scality:227c0b0...scality:3782a97#diff-8375618fe20c3e0fb1000ac2044b44b6R22

Reply:
https://github.com/scality/metalk8s/compare/improvement/GH-2498-simple-kubectl-plugin...user/nicolast/kubectl-plugin#diff-8375618fe20c3e0fb1000ac2044b44b6R22